docs: improve CRD reference intro paragraphs (#3320)
keptn/lifecycle-toolkit: docs/docs/reference/crd-reference/task.md

# KeptnTask

Keptn uses `KeptnTask` resources internally
to manage tasks (and their underlying Kubernetes Job resources)
that are run before and after deployment of your workloads  
(pre- and post-deployment tasks).
You do not need to create this resource yourself except in special situations,

> like using Keptn to manage workloads that are outside the k8s cluster.

Reviewer: ```suggestion
for instance when using Keptn to manage workloads outside the k8s cluster.
```
